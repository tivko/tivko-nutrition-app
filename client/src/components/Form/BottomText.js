import { Text, View, TouchableOpacity } from "react-native";
import React from "react";
import createStyles from "../../styles/base";
const styles = createStyles();
const BottomText = ({
  firstText,
  secondText,
  thirdText,
  pushButton,
  screenName,
}) => {
  return (
    <>
      {firstText && (
        <TouchableOpacity
          onPress={() => {
            navigation.navigate(screenName);
          }}
        >
          <Text style={[styles.text, { color: "red" }]}>{firstText}</Text>
        </TouchableOpacity>
      )}
      {secondText && (
        <View style={[styles.text, { flexDirection: "row", gap: 3 }]}>
          <Text style={styles.text}>{secondText}</Text>
          <TouchableOpacity onPress={pushButton}>
            <Text style={[styles.text, { textDecorationLine: "underline" }]}>
              {thirdText}
            </Text>
          </TouchableOpacity>
        </View>
      )}
    </>
  );
};

export default BottomText;
