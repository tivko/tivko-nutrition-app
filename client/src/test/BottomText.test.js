import React from "react";
import { TouchableOpacity } from "react-native";
import renderer from "react-test-renderer";
import BottomText from "../components/Form/BottomText";

describe("BottomText", () => {
  it("Рендериться коректно з параметром firstText", () => {
    const component = renderer.create(
      <BottomText firstText="First Text" screenName="ScreenName" />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Рендериться коректно з параметрами secondText та thirdText", () => {
    const mockPushButton = jest.fn();
    const component = renderer.create(
      <BottomText
        secondText="Second Text"
        thirdText="Third Text"
        pushButton={mockPushButton}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Викликає функцію pushButton при натисканні на thirdText", () => {
    const mockPushButton = jest.fn();
    const component = renderer.create(
      <BottomText
        secondText="Second Text"
        thirdText="Third Text"
        pushButton={mockPushButton}
      />
    );
    const instance = component.root;
    const touchableOpacity = instance.findByType(TouchableOpacity);
    expect(touchableOpacity).toBeDefined();
    if (touchableOpacity) {
      touchableOpacity.props.onPress();
    }
    expect(mockPushButton).toHaveBeenCalledTimes(1);
  });
});
