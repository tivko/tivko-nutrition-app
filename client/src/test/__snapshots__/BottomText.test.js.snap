// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomText renders correctly with firstText prop 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <Text
    style={
      [
        {
          "fontSize": 24,
        },
        {
          "color": "red",
        },
      ]
    }
  >
    First Text
  </Text>
</View>
`;

exports[`BottomText renders correctly with secondText and thirdText props 1`] = `
<View
  style={
    [
      {
        "fontSize": 24,
      },
      {
        "flexDirection": "row",
        "gap": 3,
      },
    ]
  }
>
  <Text
    style={
      {
        "fontSize": 24,
      }
    }
  >
    Second Text
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        [
          {
            "fontSize": 24,
          },
          {
            "textDecorationLine": "underline",
          },
        ]
      }
    >
      Third Text
    </Text>
  </View>
</View>
`;

exports[`BottomText Рендериться коректно з параметрами secondText та thirdText 1`] = `
<View
  style={
    [
      {
        "fontSize": 24,
      },
      {
        "flexDirection": "row",
        "gap": 3,
      },
    ]
  }
>
  <Text
    style={
      {
        "fontSize": 24,
      }
    }
  >
    Second Text
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        [
          {
            "fontSize": 24,
          },
          {
            "textDecorationLine": "underline",
          },
        ]
      }
    >
      Third Text
    </Text>
  </View>
</View>
`;

exports[`BottomText Рендериться коректно з параметром firstText 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <Text
    style={
      [
        {
          "fontSize": 24,
        },
        {
          "color": "red",
        },
      ]
    }
  >
    First Text
  </Text>
</View>
`;
